pysmt/smtlib/parser.py

```diff
@@ -514,8 +534,12 @@ def get_command_generator(self, script):
 
     def get_script_fname(self, script_fname):
         """Given a filename and a Solver, executes the solver on the file."""
-        with io.BufferedReader(io.FileIO(script_fname, 'r')) as script:
-            return self.get_script(script)
+        if PY2:
+            with io.BufferedReader(io.FileIO(script_fname, 'r')) as script:
```

Comment:
Same as above. It might be the case that this Buffered reading patter could go within pysmt.utils.